tests/test_base_plugin.py

```
@@ -832,6 +863,116 @@ async def run_test():
+            with patch("mmrelay.plugins.base_plugin.store_plugin_data") as mock_store:
+                mock_get.return_value = existing_data
+
+                # Adding new data should trigger truncation
+                new_data = {"data": "item3"}
+                plugin.store_node_data("!node123", new_data)
+
+                # The logic is: truncate existing data first, then add new data
+                # So existing_data[-2:] = existing_data (no change), then append new_data
+                expected_data = existing_data + [
+                    new_data
+                ]  # Truncate happens before append
```

Ruff (0.14.3)

Consider `[*existing_data, new_data]` instead of concatenation

(RUF005)